### PR TITLE
BinarizeParseMojo generalization

### LINKED ISSUE #2649: BinarizeParseMojo.java:145-148: We can make the current...
The puzzle `2609-cb1cc596` from #2609 has to be resolved: 

https://github.com/objectionary/eo/blob/f384b1c98c9dc66e18d160cdb4216012a6047bc2/eo-maven-plugin/src/main/java/org/eolang/maven/BinarizeParseMojo.java#L145-L148

The puzzle was created by @rultor on 29-Nov-23. 

Estimate: 90 minutes,  role: DEV. 

If you have any technical questions, don't ask me, submit new tickets instead. The task will be \"done\" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

### PR DESCRIPTION
**Pull request by c71n93**
Closes #2649

---

### Detailed summary
- The focus of this PR is to modify the `BinarizeParseMojo` class in the `eo-maven-plugin` project.
- Notable changes include:
  - Renaming the `addRust` method to `addFFIs`.
  - Updating the Javadoc comments to reflect the purpose of the class and methods.
  - Adding a todo comment to make the class more generic and remove dependencies on Rust-specific code.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: eo-maven-plugin/src/main/java/org/eolang/maven/BinarizeParseMojo.java [hunk 1]
```diff
@@ -142,14 +145,17 @@ private XML addRust(
...
-     *  by transferring this.addRust(input) snippet to corresponding
-     *  FFINode- {@link RustNode}. We wanna make the class independent of
-     *  ffi-insert as a result.
+     * @todo #2649:90min This method may be more general. We need to get rid from rust dependencies
+     *  in this method, because when adding another type of inserts it will be just copy-paste here.
+     *  First of all, the for-loop must create all kinds of FFI nodes, not only {@link RustNode}. I
+     *  think we can implement it, using something like {@code FFINodeFactory}, that will return
+     *  appropriate FFI node for every XML node from {@code nodes}. Also it will be great to move
+     *  paths to XML FFI insert nodes (such as {@code "/program/rusts/rust"}) from this method to
+     *  a class field.
```
Comment (levBagryansky) [inline eo-maven-plugin/src/main/java/org/eolang/maven/BinarizeParseMojo.java:154]: @c71n93 we can specify "static field" here

**Review by levBagryansky — APPROVED** [review]
@c71n93 please check

**Review by maxonfjvipon — APPROVED** [review]
@c71n93 LGTM! Thanks

**Comment by c71n93** [comment]
@yegor256 could you check this one, please?

**Comment by yegor256** [comment]
@rultor mertge

**Pull request merged**